In [1]:
# LIBRERIAS UTILIZADAS
import numpy as np
import pandas as pd
import gmaps
import gmaps.datasets

In [4]:
listing = pd.read_csv("df_final.csv")

In [5]:
listing["latitude"].describe()

count    13252.000000
mean        41.392081
std          0.014308
min         41.352610
25%         41.380888
50%         41.389660
75%         41.401640
max         41.461930
Name: latitude, dtype: float64

Observamos un maximo de 41.46193 y un minimo de 41.35261, por lo tanto escogeremos un intervalo de [41.30 , 41.50]

In [6]:
listing["longitude"].describe()

count    13252.000000
mean         2.167541
std          0.017475
min          2.091590
25%          2.157437
50%          2.168605
75%          2.177623
max          2.227710
Name: longitude, dtype: float64

Observamos un maximo de 2.22771 y un minimo de 2.09159, por lo tanto escogeremos un intervalo de [2.03, 2.28]

In [ ]:
gmaps.configure(api_key="AIzaSyD2q-mG8Phl5JljfPqk0Wtv4fWQrH_dvas")

Empezamos realizando un heatmap para una variable que contenga el número de viviendas que hay en alquiler.

In [25]:
Lon = np.arange(2.03, 2.28, 0.0025) 
Lat = np.arange(41.30, 41.50, 0.00201) 
listing_counts = np.zeros((100,100)) 

In [26]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_counts[b1,b2] += 1

In [28]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_counts.resize((10000,)) 

In [29]:
heatmap_data = {'Counts': listing_counts, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [30]:
locations = df[['latitude', 'longitude']] 
weights = df['Counts'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))

Haremos ahora un gráfico para la media de precio.

In [22]:
listing_price = np.zeros((100,100))

In [24]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_price[b1,b2] += listing["price"].values[a]

In [27]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_price.resize((10000,)) 

In [40]:
listing_mean_price = listing_price/listing_counts
listing_mean_price[np.isnan(listing_mean_price)] = 0

C:\Users\Andreu\AppData\Local\Temp\ipykernel_9160\2118153100.py:1: RuntimeWarning: invalid value encountered in true_divide
  listing_mean = listing_price/listing_counts


In [41]:
heatmap_data = {'Price': listing_mean_price, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [42]:
locations = df[['latitude', 'longitude']] 
weights = df['Price'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))

Finalmente realizaremos mapas de calor para la media de las dos variables respuesta, "days_booked" y "reviews_diff". Empezamos por "days_booked".

In [43]:
listing_days = np.zeros((100,100))

In [44]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_days[b1,b2] += listing["days_booked"].values[a]

In [45]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_days.resize((10000,)) 

In [46]:
listing_mean_days = listing_days/listing_counts
listing_mean_days[np.isnan(listing_mean_days)] = 0

C:\Users\Andreu\AppData\Local\Temp\ipykernel_9160\1967739511.py:1: RuntimeWarning: invalid value encountered in true_divide
  listing_mean_days = listing_days/listing_counts


In [47]:
heatmap_data = {'days booked': listing_mean_days, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [48]:
locations = df[['latitude', 'longitude']] 
weights = df['days booked'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))

Realizamos ahora para "reviews_diff".

In [90]:
listing_reviews = np.zeros((100,100))

In [91]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_reviews[b1,b2] += listing["reviews_diff"].values[a]

In [92]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_reviews.resize((10000,)) 
listing_reviews[listing_reviews<0] = 0

In [93]:
listing_mean_reviews = listing_reviews/listing_counts
listing_mean_reviews[np.isnan(listing_mean_reviews)] = 0
listing_reviews.max()

C:\Users\Andreu\AppData\Local\Temp\ipykernel_9160\3517011110.py:1: RuntimeWarning: invalid value encountered in true_divide
  listing_mean_reviews = listing_reviews/listing_counts


429.0

In [84]:
heatmap_data = {'reviews diff': listing_mean_reviews, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [85]:
locations = df[['latitude', 'longitude']] 
weights = df['reviews diff'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))